### PR TITLE
Release template: Document tagging for API submodule

### PR DESCRIPTION
**Pull request by ArangoGutierrez**

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
Last one for the V0.17 Milestone @marquiz

File: .github/ISSUE_TEMPLATE/new-patch-release.md [hunk 1]
```diff
@@ -19,8 +19,12 @@ Please do not remove items from the checklist
   - [ ] Upload release artefacts generated by `prepare-release.sh` script above to the draft release
 - [ ] An OWNER runs
      `git tag -s -m "NFD release $VERSION" $VERSION`
+- [ ] An OWNER runs
+     `cd api/nfd`
```
Comment (marquiz) [inline .github/ISSUE_TEMPLATE/new-patch-release.md:23]: Should be able to create the tag in the repo root

```suggestion
- [ ] An OWNER runs
     `cd api/nfd`
```

Comment (ArangoGutierrez) [reply .github/ISSUE_TEMPLATE/new-patch-release.md]: Done

File: .github/ISSUE_TEMPLATE/new-patch-release.md [hunk 2]
```diff
@@ -19,8 +19,11 @@ Please do not remove items from the checklist
   - [ ] Upload release artefacts generated by `prepare-release.sh` script above to the draft release
 - [ ] An OWNER runs
      `git tag -s -m "NFD release $VERSION" $VERSION`
+- [ ] An OWNER runs
+     `git tag -s -m "NFD API release $VERSION" api/nfd/$VERSION`     
 - [ ] An OWNER pushes the tag with
       `git push $VERSION`
+      `git push api/nfd/$VERSION`
```
Comment (marquiz) [inline .github/ISSUE_TEMPLATE/new-patch-release.md:26]: push both tags with a single command

```suggestion
      `git push $VERSION api/nfd/$VERSION`
```

Alternatively have two separate checklist items like for the tagging above. WDYT?

Comment (ArangoGutierrez) [reply .github/ISSUE_TEMPLATE/new-patch-release.md]: Done

**Review by marquiz — APPROVED** [review]
Thank you @ArangoGutierrez
/lgtm

**Pull request merged**